lib/cartopy/img_transform.py

```diff
@@ -270,15 +275,20 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
                                            target_x_points.flatten(),
                                            target_y_points.flatten())
 
-    # Versions of scipy >= v0.16 added the balanced_tree argument,
-    # which caused the KDTree to hang with this input.
-    try:
-        kdtree = scipy.spatial.cKDTree(xyz, balanced_tree=False)
-    except TypeError:
-        kdtree = scipy.spatial.cKDTree(xyz)
-
-    distances, indices = kdtree.query(target_xyz, k=1)
-    mask = np.isinf(distances)
+    if _is_pykdtree:
+        kdtree = pykdtree.kdtree.KDTree(xyz)
+        # Use sqr_dists=True because we don't care about distances,
+        # and it saves a sqrt.
+        _, indices = kdtree.query(target_xyz, k=1, sqr_dists=True)
+    else:
+        # Versions of scipy >= v0.16 added the balanced_tree argument,
+        # which caused the KDTree to hang with this input.
+        try:
+            kdtree = scipy.spatial.cKDTree(xyz, balanced_tree=False)
+        except TypeError:
+            kdtree = scipy.spatial.cKDTree(xyz)
+        _, indices = kdtree.query(target_xyz, k=1)
+    mask = indices >= len(xyz)
```

Comment:
This change is there because we don't really care about the distance and pykdtree returns differing results for out-of-domain vs. nan https://github.com/storpipfugl/pykdtree/issues/35